lib/engine/game/g_1849.rb

```diff
@@ -39,6 +39,41 @@ class G1849 < Base
         phase_limited: :blue,
       }.freeze
 
+      def track_type(paths)
+        types = paths.map(&:track).uniq
+        game_error('Can only change track type at station.') if types.include?(:broad) && types.include?(:narrow)
+        case
+        when types.include?(:narrow)
+          :narrow
+        when types.include?(:broad)
+          :broad
+        else
+          :dual
+        end
+      end
+
+      def hex_edge_cost(conn, train)
+        track = track_type(conn.paths)
+        edges = conn.paths.each_cons(2).sum do |pair|
```

Comment:
you could do |a, b|

Reply:
I'd also suggest putting this into a common hex_train.rb class as 18Ireland will use it as well as other hex based games in the future.

Reply:
I originally had a new `Train` class but trains don't do any of this computation - games and routes do.